downstream.go

```diff
@@ -1077,6 +1077,46 @@ func (dc *downstreamConn) handleMessageRegistered(msg *irc.Message) error {
 			}
 			sendTopic(dc, ch)
 		}
+	case "LIST":
+		// TODO: support ELIST when supported by all upstreams
+
+		dc.user.pendingListsLock.Lock()
+		defer dc.user.pendingListsLock.Unlock()
+
+		pl := pendingList{
+			downstreamID:    dc.id,
+			pendingRequests: make(map[int64][]string),
+		}
+		var upstreamChannels map[int64][]string
+		if len(msg.Params) > 0 {
+			upstreamChannels = make(map[int64][]string)
+			channels := strings.Split(msg.Params[0], ",")
+			for _, channel := range channels {
+				uc, upstreamChannel, err := dc.unmarshalEntity(channel)
+				if err != nil {
+					return err
+				}
+				if c, ok := upstreamChannels[uc.network.ID]; ok {
+					upstreamChannels[uc.network.ID] = append(c, upstreamChannel)
+				} else {
+					upstreamChannels[uc.network.ID] = []string{upstreamChannel}
+				}
+			}
+		}
+
+		dc.user.pendingLists = append(dc.user.pendingLists, pl)
+		dc.forEachUpstream(func(uc *upstreamConn) {
+			var params []string
+			if upstreamChannels == nil {
+				params = []string{}
```

Comment:
Nit: no need to initialize `params`, `params = nil` would do (or just do nothing).